tests/LDContextBuilderTest.php

```diff
@@ -0,0 +1,358 @@
+                'key',
+                'value',
+                new LDContext(
+                    LDContext::DEFAULT_KIND,
+                    'value',
+                    null,
+                    false,
+                    null,
+                    null,
+                    null,
+                    null
+                )
+            ],
+            'key not string' => [
+                false,
+                'key',
+                12345,
+                new LDContext(
+                    LDContext::DEFAULT_KIND,
+                    'some-context-key',
+                    null,
+                    false,
+                    null,
+                    null,
+                    null,
+                    null
+                )
+            ],
+            'kind string' => [
+                true,
+                'kind',
+                'value',
+                new LDContext(
+                    'value',
+                    'some-context-key',
+                    null,
+                    false,
+                    null,
+                    null,
+                    null,
+                    null
+                )
+            ],
+            'kind not string' => [
+                false,
+                'kind',
+                12345,
+                new LDContext(
+                    LDContext::DEFAULT_KIND,
+                    'some-context-key',
+                    null,
+                    false,
+                    null,
+                    null,
+                    null,
+                    null
+                )
+            ],
+            'anonymous bool' => [
+                true,
+                'anonymous',
+                true,
+                new LDContext(
+                    LDContext::DEFAULT_KIND,
+                    'some-context-key',
+                    null,
+                    true,
+                    null,
+                    null,
+                    null,
+                    null
+                )
+            ],
+            'anonymous not bool' => [
+                false,
+                'anonymous',
+                12345,
+                new LDContext(
+                    LDContext::DEFAULT_KIND,
+                    'some-context-key',
+                    null,
+                    false,
+                    null,
+                    null,
+                    null,
+                    null
+                )
+            ],
+            'attributes value not null' => [
+                true,
+                'attribute-name',
+                'value',
+                new LDContext(
+                    LDContext::DEFAULT_KIND,
+                    'some-context-key',
+                    null,
+                    false,
+                    ['attribute-name' => 'value'],
+                    null,
+                    null,
+                    null
+                )
+            ],
+            'attributes name value' => [
```

Comment:
We don't have a test checking setting a custom attribute to null. You could probably just fold that into the `testBuilderSetCustomAttributes` test.